paasta_tools/cli/cmds/spark_run.py

```diff
@@ -606,6 +652,52 @@ def get_spark_app_name(original_docker_cmd: Union[Any, str, List[str]]) -> str:
     return spark_app_name
 
 
+def _calculate_docker_memory_limit(
+    spark_conf: Mapping[str, str], memory_limit: Optional[str]
+) -> str:
+    """ In Order of preference:
+    1. Argument: --docker-memory-limit
+    2. --spark-args or spark-submit: spark.driver.memory
+    3. Default
+    """
+    if memory_limit:
+        return memory_limit
+
+    try:
+        docker_memory_limit_str = spark_conf.get(
+            "spark.driver.memory", DEFAULT_DRIVER_MEMORY_BY_SPARK
+        )
+        adjustment_factor = DOCKER_RESOURCE_ADJUSTMENT_FACTOR
+        match = re.match(r"([0-9]+)([a-z]*)", docker_memory_limit_str)
+        memory_val = int(match[1]) * adjustment_factor
+        memory_unit = match[2]
+        docker_memory_limit = f"{memory_val}{memory_unit}"
+    except Exception as e:
+        # For any reason it fails, continue with default value
+        print(
+            f"Failed to parse docker memory limit. Error: {e}. Example values: 1g, 200m."
+            " Setting a default value {DEFAULT_MAX_DOCKER_MEMORY_LIMIT}"
+        )
+        docker_memory_limit = DEFAULT_MAX_DOCKER_MEMORY_LIMIT
```

Comment:
thinking about this more: should we just fail here? we're already going to fail if people do something weird with the cpu limit too

Reply:
Most users specify high memory limit as part of over-provisioning for successful run of the job.
Having it failed IMHO is not a smooth user-experience for both onpoint and end-user (delay, notice, reach onpoint, change memory limits, calculate the right value, knowing the max value).

Even if it fails, the user would need to specify the max driver limit of the container (or half of it). 
Having it as a warning is good enough to be aware of and take decisions if required. 
I would rather prefer to keep it like it this.

Reply:
this block would only get hit if users made a mistake setting up their config tho, right? e.g., if they set `10h` instead of `10g`, we/they would never know that their config is invalid and this would basically mean that we fall back to the status quo where a single spark job can take down a box

Reply:
Sorry I misinterpreted your comment in lines with failing when user specifies > than the max allowed limit of the box.
Yes, you are right.
My original reasoning to not fail was, in the goal of not impacting batches doesn't fall into this regex and keep the job running as status-quo. As in WARN, Move-on, fix based on warning for next runs.

I guess I will change this to fail-hard.

Reply:
would it be worth writing a quick script to check all the existing configs to see if there are any existing batches that have incorrect config?